src/hardware/gus.cpp

```diff
@@ -1443,26 +1468,42 @@ void Gus::WriteToPort(io_port_t port, io_val_t value, io_width_t width)
 		// TODO Check if 0x20a register is also available on the gus
 		// like on the interwave
 	case 0x20b:
-		if (!should_change_irq_dma) {
-			break;
-		}
-		should_change_irq_dma = false;
-		if (mix_ctrl & 0x40) {
-			// IRQ configuration, only use low bits for irq 1
-			const auto i        = val & 7;
-			const auto& address = irq_addresses.at(i);
-			if (address) {
-				irq1 = address;
-			}
+		if (should_change_irq_dma) {
+
+			//  The write to 2XB MUST occur as the NEXT IOW or else
+			//  the write to 2XB will be locked out and not occur.
+			//  This is to prevent an application that is probing
+			//  for cards from accidentally corrupting the latches.
+			//  UltraSound Software Development Kit (SDK),
+			//  Section 2.13.
+			//
+			should_change_irq_dma = false;
+
+			// The address selector is set in bits 2-0
+			const auto selector = static_cast<uint8_t>(val & 0b111);
+
+			// When Bit 6 (Control Register Select) is set to a 1,
+			// the next IO write to 2XB will be to the IRQ control
+			// latches. When this is set to a 0, the next IO write
+			// to 2XB will be to the DMA channel latches. UltraSound
+			// Software Development Kit (SDK), Section 2.13.
+			//
+			const auto is_irq_selected = bit::is(mix_ctrl,
+			                                     bit::literals::b6);
```

Comment:
Next level: `bit_view` register definitions 😉 😉 😆

Reply:
Ahh yes; even better. 
I see that `mix_ctrl` isn't used that much, so this will fit fine with this PR (without treading into wide-scale refactor territory).

Reply:
`bit_view`'d the mix control and address selector regs :+1: that's a lot more self documenting now :-)

Reply:
> `bit_view`'d the mix control and address selector regs 👍 that's a lot more self documenting now :-)

Noice 😎